File: queue/queue.js
var RecentCounter = function() {
    this.queue = []
};

/** 
 * @param {number} t
 * @return {number}
 */
RecentCounter.prototype.ping = function(t) {

    if(t == null) return null
    while(true){
        if((t-this.queue[0]) > 3000){
            this.queue.shift()
        }else{
            break
        }
    }
    this.queue.push(t)
    return this.queue.length

};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * var obj = new RecentCounter()
 * var param_1 = obj.ping(t)
 */
var obj = new RecentCounter()
console.log(obj.ping(null))
console.log(obj.ping(642))
console.log(obj.ping(1849))
console.log(obj.ping(4921))
console.log(obj.ping(5936))
console.log(obj.ping(5957))

File: queue/queue.py
from collections import deque
class RecentCounter:
    
    def __init__(self):
        self.queue = deque()
        

    def ping(self, t: int) -> int:
        if t is None:
            return None
        while True:
            if len(self.queue) and ((t-self.queue[0]) > 3000):
                self.queue.popleft()
            else:
                break
        self.queue.append(t)
        print(self.queue)
        return len(self.queue)



# Your RecentCounter object will be instantiated and called as such:
obj = RecentCounter()
print(obj.ping(None))
print(obj.ping(642))
print(obj.ping(1849))
print(obj.ping(4921))
print(obj.ping(5936))
print(obj.ping(5957))